samples/matter/common/src/bridge/ble_connectivity_manager.cpp

```diff
@@ -257,6 +261,9 @@ void BLEConnectivityManager::PairingComplete(struct bt_conn *conn, bool bonded)
 		return;
 	}
 
+	/* Indicate that pairing has been finished */
+	Instance().NotifyUpdateState(State::Idle);
```

Comment:
Generally I'm not sure if this notify-callback architecture will work correctly. My understanding is that this is quite generic mechanism that could be used for anything (in our particular case for blinking LEDs).

In the current implementation transition between the states is done in a way that sometimes give false information to the user (for LED use case it does not matter, but generally for someone it could matter). E.g. let's assume we have one device and we start adding new one, it will go like this:

LostDevice->Scanning->Pairing->**Idle**->LostDevice

So at some point the user will be notified that bridge is in the idle state, while it's not true, because it should be in `LostDevice`.